src/r2mm/downloading/BetterThunderstoreDownloader.ts

```diff
@@ -145,10 +145,12 @@ export default class BetterThunderstoreDownloader extends ThunderstoreDownloader
 
         await this.buildDependencySet(combo.getVersion(), dependencies, versionMode);
         this.sortDependencyOrder(dependencies);
-        // #270: Remove already-installed dependencies to prevent updating.
-        return dependencies.filter((dep) => {
-            return !(modList.some(installed => installed.getName() === dep.getMod().getFullName()));
-        });
+
+        return isModpack ? dependencies :
+            // #270: Remove already-installed dependencies to prevent updating.
```

Comment:
It might be clearer if we dropped this comment altogether and replaced the docstring on L140 with something like:

> When installing a modpack ensure compatibility by installing the exact
> versions of dependencies even if it means upgrading/downgrading already
> installed mods. When installing a regular mod, install the latest
> versions of dependencies but don't change already installed mods.

Reply:
Comment: Updated the comment based on the comment.